### ISSUE TITLE
time_cut

### ISSUE THREAD

**Issue by nicholebarry** [EoRImaging/FHD#73]
Looks like the time_cut keyword is only applied if flag_visibilities is set (as in, time_cut is only applied in vis_flag which is only called if flag_visibilities is set).

**Comment by nicholebarry** [comment]
I set flag_visibilities and it still wasn't used...don't know what's going on here

[1061313496_settings.txt](https://github.com/EoRImaging/FHD/files/883464/1061313496_settings.txt)
[firstpass.o76738.1.txt](https://github.com/EoRImaging/FHD/files/883465/firstpass.o76738.1.txt)

**Comment by isullivan** [comment]
See pull request #75 . Please try that branch, and merge the pull request if it works.

**Comment by nicholebarry** [comment]
Time_cut seems to be working well. Only issue now is the noise calculation...the splitting of the even-odd sets, in particular. 

When the time cut is applied to the beginning and end of the observation (cut one step at the front, one step at the back), the even-odd sets appears off. I can tell this is the case by looking at the noise ratio plots (sorry I don't have plots....that's another story). However, if two time steps are cut at the end of the observation, the noise ratio plots look like the even-odd sets were calculated correctly.

**Comment by bhazelton** [comment]
It sounds like maybe the definition what is even or odd is not being adjusted based on time flags.

**Comment by isullivan** [comment]
Yes, "even" and "odd" start from the first non-flagged time, to ensure that we use as much data as possible and that the same amount of data goes into even and odd visibilities. This means that if you flag the first two seconds (one time step, typically), the "even" visibilities will start from index 1 instead of 0 and the odd visibilities will start from index 2 instead of 1.

**Comment by bhazelton** [comment]
That sounds like the right thing to do, but I think Nichole is seeing signal leak into the noise ratio plots in this case. This indicates that the even and odd cubes aren't perfectly matched, which is sometime caused by flagging differences. So I suspect that something isn't happening quite right with the flag matching in this situation.

**Comment by isullivan** [comment]
That's of course possible, though that will take a bit of debugging to sort out. I think I'd want to see the plots and any diagnostic output first before trying to hunt that down.
